pkg/scheduler/framework/plugins/schedulinggates/scheduling_gates.go

```diff
@@ -50,10 +50,15 @@ func (pl *SchedulingGates) PreEnqueue(ctx context.Context, p *v1.Pod) *framework
 	return framework.NewStatus(framework.UnschedulableAndUnresolvable, fmt.Sprintf("waiting for scheduling gates: %v", gates))
 }
 
-// EventsToRegister returns nil here to indicate that schedulingGates plugin is not
-// interested in any event but its own update.
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
 func (pl *SchedulingGates) EventsToRegister() []framework.ClusterEventWithHint {
-	return nil
+	return []framework.ClusterEventWithHint{
+		// Pods can be more schedulable once it's gates are removed
+		// We don't have QueueingHint intentionally because QueueingHint would be a very similar logic as PreEnqueue,
+		// and it wouldn't add any value to the efficiency.
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Update}},
```

Comment:
We'll have a performance degradation here since we'll listen for all pod update events now but before we only listen for self updates. That means you should add a hint func.

Reply:
https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1421611944
We'll run other plugins' QHint anyway, and the preEnqueue logic and QHint logic for `scheduling_gate` is the same. So I think they have the same calculation cost?

1. With QHint, run (n+1) QHint functions (other plugins + gate plugin)
2. Without QHint, run n QHint functions and preEnqueue logic.

Reply:
At least, no need to delete and add back to the unschedulableQ again and again, and what if we have several preEnqueue plugins? No need to run the whole preEnqueue logic.